### PR TITLE
DOC: Add checklist to PR template

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add checklist to PR template to:
- Remind users about the necessary parts for a good PR.
- Allow reviewers get a grasp of the PR at a glance.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Inspired by:
https://github.com/PowerShell/PowerShell/blob/master/.github/PULL_REQUEST_TEMPLATE.md

The benefit of this is arguable in terms of the additional overhead, so I'll be fine if you decide to decline the PR.

If we accept the change, it references some of the sections added in PR #627, so we may need to adapt it/rebase it on master if the lattter gets merged.

**Comment by jhlegarreta** [comment]
6243b8b rebased on master.

**Comment by mystrykiller** [comment]
could you help me get a copy of the master ? got to love grand children

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -27,4 +27,17 @@ close a related issues using keywords (https://help.github.com/articles/closing-
 @mentions (https://help.github.com/articles/basic-writing-and-formatting-syntax/#mentioning-people-and-teams)
 of the person or team responsible for reviewing proposed changes. -->
 
+## PR Checklist
+- [ ] [Makes breaking changes](https://raw.githubusercontent.com/InsightSoftwareConsortium/ITK/master/CONTRIBUTING.md#making-breaking-changes)
```
Comment (thewtex) [inline .github/pull_request_template.md]: *making-breaking-changes* -> *breaking-changes*

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -27,4 +27,17 @@ close a related issues using keywords (https://help.github.com/articles/closing-
...
+- [ ] [Makes design changes](https://raw.githubusercontent.com/InsightSoftwareConsortium/ITK/master/CONTRIBUTING.md#making-design-changes)
```
Comment (thewtex) [inline .github/pull_request_template.md]: *making-design-changes* ->*design-changes*

Comment (jhlegarreta) [reply .github/pull_request_template.md]: :eyes: Done in 453f1e9. Thanks !

File: .github/pull_request_template.md (hunk 1 above)
Comment (jhlegarreta) [reply .github/pull_request_template.md]: :eyes: Done in 453f1e9. Thanks !

**Pull request merged**